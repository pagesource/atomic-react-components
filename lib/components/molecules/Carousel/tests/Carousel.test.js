import React from 'react';
import { mountWithTheme } from '../../../../../__mock__/themeMock';

import { CarouselVanilla } from '../index';

describe('<Carousel />', () => {
  let CarouselComponent = '';

  beforeEach(() => {
    CarouselComponent = mountWithTheme(
      <CarouselVanilla>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 4 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 2 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
      </CarouselVanilla>
    );
  });

  it('should render correctly', () => {
    expect(CarouselComponent).toMatchSnapshot();
  });

  it('should have slider wrapper as one child only', () => {
    expect(CarouselComponent.find('.carousel-section')).toHaveLength(1);
  });

  it('Children should have first slide as active initially', () => {
    expect(
      CarouselComponent.find('.carousel-section')
        .find('.slide')
        .at(0)
        .hasClass('active')
    ).toEqual(true);
    expect(
      CarouselComponent.find('.carousel-section')
        .find('.slide')
        .at(1)
        .hasClass('active')
    ).toEqual(false);
  });

  it('Should display dots on using <Carousel.Dots> component only', () => {
    let DotsComponent = mountWithTheme(
      <CarouselVanilla>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
      </CarouselVanilla>
    );
    expect(DotsComponent.exists('.carousel-dots')).toBe(false);
    DotsComponent = mountWithTheme(
      <CarouselVanilla>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
        <CarouselVanilla.Dots />
      </CarouselVanilla>
    );
    expect(DotsComponent.exists('.carousel-dots')).toBe(true);
  });

  it('Should change slide on click dots', () => {
    const CarouselWithDots = mountWithTheme(
      <CarouselVanilla>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 2 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 4 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
        <CarouselVanilla.Dots />
      </CarouselVanilla>
    );
    CarouselWithDots.find('.carousel-dots')
      .find('button')
      .at(3)
      .simulate('click');
    expect(
      CarouselWithDots.find('.carousel-section')
        .find('.slide')
        .at(3)
        .hasClass('active')
    ).toEqual(true);
  });

  it('Should active next slide on click next', () => {
    const CarouselWithDots = mountWithTheme(
      <CarouselVanilla navigation>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 2 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 4 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
        <CarouselVanilla.Dots />
      </CarouselVanilla>
    );
    CarouselWithDots.find('.next')
      .find('button')
      .simulate('click');
    expect(
      CarouselWithDots.find('.carousel-section')
        .find('.slide')
        .at(1)
        .hasClass('active')
    ).toEqual(true);
  });

  it('Should active previous slide on key press right arrow', () => {
    const CarouselWithDots = mountWithTheme(
      <CarouselVanilla navigation>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 2 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 4 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
        <CarouselVanilla.Dots />
      </CarouselVanilla>
    );
    CarouselWithDots.find('.next')
      .find('button')
      .simulate('keyup', { key: 'ArrowRight' });
    expect(
      CarouselWithDots.find('.carousel-section')
        .find('.slide')
        .at(1)
        .hasClass('active')
    ).toEqual(true);
  });

  it('Should active previous slide on click previous', () => {
    const CarouselWithDots = mountWithTheme(
      <CarouselVanilla navigation>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 2 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 4 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
        <CarouselVanilla.Dots />
      </CarouselVanilla>
    );
    CarouselWithDots.find('.next')
      .find('button')
      .simulate('click');
    CarouselWithDots.find('.next')
      .find('button')
      .simulate('click');
    CarouselWithDots.find('.prev')
      .find('button')
      .simulate('click');
    expect(
      CarouselWithDots.find('.carousel-section')
        .find('.slide')
        .at(1)
        .hasClass('active')
    ).toEqual(true);
  });

  it('Should active previous slide on key press left', () => {
    const CarouselWithDots = mountWithTheme(
      <CarouselVanilla navigation>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 3 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 2 </CarouselVanilla.Slide>
          <CarouselVanilla.Slide>TEst 4 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
        <CarouselVanilla.Dots />
      </CarouselVanilla>
    );
    CarouselWithDots.find('.next')
      .find('button')
      .simulate('keyup', { key: 'ArrowRight' });
    CarouselWithDots.find('.next')
      .find('button')
      .simulate('keyup', { key: 'ArrowRight' });
    CarouselWithDots.find('.prev')
      .find('button')
      .simulate('keyup', { key: 'ArrowLeft' });
    expect(
      CarouselWithDots.find('.carousel-section')
        .find('.slide')
        .at(1)
        .hasClass('active')
    ).toEqual(true);
  });

  it('Should display Navigation extract from Component', () => {
    let CarouselWithNav = mountWithTheme(
      <CarouselVanilla nav={false}>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
      </CarouselVanilla>
    );
    expect(CarouselWithNav.exists('.prev')).toBe(false);
    expect(CarouselWithNav.exists('.next')).toBe(false);

    CarouselWithNav = mountWithTheme(
      <CarouselVanilla nav={false}>
        <CarouselVanilla.Prev>prev</CarouselVanilla.Prev>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
      </CarouselVanilla>
    );
    expect(CarouselWithNav.exists('.prev')).toBe(true);
    expect(CarouselWithNav.exists('.next')).toBe(false);

    CarouselWithNav = mountWithTheme(
      <CarouselVanilla nav={false}>
        <CarouselVanilla.SlideWrapper>
          <CarouselVanilla.Slide>TEst 1 </CarouselVanilla.Slide>
        </CarouselVanilla.SlideWrapper>
        <CarouselVanilla.Next>next</CarouselVanilla.Next>
      </CarouselVanilla>
    );
    expect(CarouselWithNav.exists('.prev')).toBe(false);
    expect(CarouselWithNav.exists('.next')).toBe(true);
  });
});
