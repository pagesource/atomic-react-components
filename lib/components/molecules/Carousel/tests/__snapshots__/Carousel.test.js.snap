// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> should render correctly 1`] = `
<Carousel
  autoplay={false}
  controls={false}
  dots={true}
  itemsPerSlide={1}
  responsive={Array []}
  slideAnimationTime={3000}
  slideTransitionTime={2000}
>
  <section
    className="carousel-wrapper"
  >
    <Component>
      <div
        className="carousel-section"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="carousel-sliding-wrapper"
        >
          <Slide
            id={0}
            key=".0"
          >
            <div
              className="slide active"
            >
              TEst 1 
            </div>
          </Slide>
          <Slide
            id={1}
            key=".1"
          >
            <div
              className="slide"
            >
              TEst 3 
            </div>
          </Slide>
          <Slide
            id={2}
            key=".2"
          >
            <div
              className="slide"
            >
              TEst 4 
            </div>
          </Slide>
          <Slide
            id={3}
            key=".3"
          >
            <div
              className="slide"
            >
              TEst 2 
            </div>
          </Slide>
        </div>
      </div>
    </Component>
    <ListWrapper>
      <div
        className="carousel-dots"
      >
        <List__StyledList>
          <List
            ListType="ul"
            borderSeparator={false}
            className="List__StyledList-rp20fd-0 dlvIbB"
            inheritedStyles=""
            inline={false}
            nested={false}
            noDefaultView={false}
            noMargin={false}
          >
            <ul
              borderSeparator={false}
              className="List__StyledList-rp20fd-0 dlvIbB"
              inheritedStyles=""
              inline={false}
              nested={false}
              noDefaultView={false}
              noMargin={false}
            >
              <GetList>
                <li
                  className="item active"
                  key="0"
                >
                  <Button__StyledButton
                    className="icon"
                    onClick={[Function]}
                  >
                    <Button
                      ariaLabel=""
                      className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                      disabled={false}
                      elementLocator=""
                      inheritedStyles=""
                      onClick={[Function]}
                      primary={false}
                      secondary={false}
                      tertiary={false}
                      transparent={false}
                      type="button"
                      variation=""
                    >
                      <button
                        aria-label={null}
                        className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                        data-locator="button-button-"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                        variation=""
                      >
                        1
                      </button>
                    </Button>
                  </Button__StyledButton>
                </li>
                <li
                  className="item"
                  key="1"
                >
                  <Button__StyledButton
                    className="icon"
                    onClick={[Function]}
                  >
                    <Button
                      ariaLabel=""
                      className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                      disabled={false}
                      elementLocator=""
                      inheritedStyles=""
                      onClick={[Function]}
                      primary={false}
                      secondary={false}
                      tertiary={false}
                      transparent={false}
                      type="button"
                      variation=""
                    >
                      <button
                        aria-label={null}
                        className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                        data-locator="button-button-"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                        variation=""
                      >
                        2
                      </button>
                    </Button>
                  </Button__StyledButton>
                </li>
                <li
                  className="item"
                  key="2"
                >
                  <Button__StyledButton
                    className="icon"
                    onClick={[Function]}
                  >
                    <Button
                      ariaLabel=""
                      className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                      disabled={false}
                      elementLocator=""
                      inheritedStyles=""
                      onClick={[Function]}
                      primary={false}
                      secondary={false}
                      tertiary={false}
                      transparent={false}
                      type="button"
                      variation=""
                    >
                      <button
                        aria-label={null}
                        className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                        data-locator="button-button-"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                        variation=""
                      >
                        3
                      </button>
                    </Button>
                  </Button__StyledButton>
                </li>
                <li
                  className="item"
                  key="3"
                >
                  <Button__StyledButton
                    className="icon"
                    onClick={[Function]}
                  >
                    <Button
                      ariaLabel=""
                      className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                      disabled={false}
                      elementLocator=""
                      inheritedStyles=""
                      onClick={[Function]}
                      primary={false}
                      secondary={false}
                      tertiary={false}
                      transparent={false}
                      type="button"
                      variation=""
                    >
                      <button
                        aria-label={null}
                        className="Button__StyledButton-sc-1ccn477-0 isjRmA icon"
                        data-locator="button-button-"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                        variation=""
                      >
                        4
                      </button>
                    </Button>
                  </Button__StyledButton>
                </li>
              </GetList>
            </ul>
          </List>
        </List__StyledList>
      </div>
    </ListWrapper>
  </section>
</Carousel>
`;
